compiler/circle-mpqsolver/src/core/Dumper.cpp

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Dumper.h"
+
+#include <luci/CircleExporter.h>
+#include <luci/CircleFileExpContract.h>
+
+#include <json.h>
+#include <fstream>
+#include <sys/stat.h>
+
+using namespace mpqsolver::core;
+
+namespace
+{
+const std::string default_dtype_key = "default_quantization_dtype";
+const std::string default_granularity_key = "default_granularity";
+const std::string layers_key = "layers";
+const std::string model_key = "model_path";
+const std::string layer_name_key = "name";
+const std::string layer_dtype_key = "dtype";
+const std::string layer_granularity_key = "granularity";
+} // namespace
+
+Dumper::Dumper(const std::string &dir_path) : _dir_path(dir_path) {}
+
+void Dumper::set_model_path(const std::string &model_path) { _model_path = model_path; }
+
+void Dumper::dump_MPQ_configuration(const LayerParams &layers, const std::string &def_dtype,
+                                    const std::string &path) const
+{
+  Json::Value mpq_data;
+  mpq_data[default_dtype_key] = def_dtype;
+  mpq_data[default_granularity_key] = "channel";
+  mpq_data[model_key] = _model_path;
+
+  Json::Value layers_data;
+  for (auto &layer : layers)
+  {
+    Json::Value layer_data;
+    layer_data[layer_name_key] = layer->name;
+    layer_data[layer_granularity_key] = layer->granularity;
+    layer_data[layer_dtype_key] = layer->dtype;
+    layers_data.append(layer_data);
+  }
+  mpq_data[layers_key] = layers_data;
+
+  Json::StreamWriterBuilder builder;
+  auto data = Json::writeString(builder, mpq_data);
+
+  write_data_to_file(path, data);
+}
+
+void Dumper::prepare_directory(const std::string &dir_path) const
+{
+  struct stat sb;
+  if (stat(dir_path.c_str(), &sb) != 0 || !S_ISDIR(sb.st_mode))
+  {
+    if (mkdir(dir_path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) != 0)
+    {
+      throw std::runtime_error("Failed to create directory for dumping intermediate results");
+    }
+  }
+}
+
+void Dumper::dump_MPQ_configuration(const LayerParams &layers, const std::string &def_dtype,
+                                    int param) const
```

Comment:
Better to rename `param`. Maybe `step` or `iter_num`?
```suggestion
                                    int step) const
```